bot/skills/roll.py

```diff
@@ -15,10 +19,67 @@
 NUM_BULLETS = 6
 
 
+class DB:
+    """
+        Hussar document:
+        {
+            _id: 420,                                   # int       -- tg user id
+            meta: {...},                                # Dict      -- full tg user object (just in case)
+            shot_counter: 10,                           # int       -- full amount of shots
+            miss_counter: 8,                            # int       -- amount of miss
+            dead_counter: 2,                            # int       -- amount of dead shots
+            total_time_in_club: datetime(...),          # DateTime  -- all time in the club
+            first_shot": datetime(...),                 # DateTIme  -- time of first shot
+            "last_shot": datetime(...)                  # DateTIme  -- time of last shot
+        }
+    """
+
+    def __init__(self, db_name: str):
+        self._coll: Collection = get_db(db_name).hussars
+
+    def find_all(self):
+        return list(self._coll.find({}))
+
+    def find(self, user_id: str):
+        return self._coll.find_one({"_id": user_id})
+
+    def add(self, user: User):
+        now: datetime = datetime.now()
+        return self._coll.insert_one({
+            "_id": user.id,
+            "meta": user.to_dict(),
+            "shot_counter": 0,
+            "miss_counter": 0,
+            "dead_counter": 0,
+            "total_time_in_club": timedelta().seconds,
+            # TODO: wat about ranks?
+            # "rank": meh,
+            "first_shot": now,
+            "last_shot": now,
+        })
+
+    def dead(self, user_id: str, until: datetime):
+        ban_duration = until - datetime.now()
+        self._coll.update_one({"_id": user_id}, {
+            "$inc": {"shot_counter": 1, "dead_counter": 1, "total_time_in_club": ban_duration.seconds},
+            "$set": {"last_shot": datetime.now()},
+        })
+
+    def miss(self, user_id: str):
+        self._coll.update_one({"_id": user_id}, {
+            "$inc": {"shot_counter": 1, "miss_counter": 1},
+            "$set": {"last_shot": datetime.now()},
+        })
+
+
+_db = DB(db_name='roll')
```

Comment:
I think we should add option to erase own roll data to be GDPR compliant 🧐

Reply:
`/roll_me_out` will remove all the data related to the current user

Reply:
Excellent! But I'd call it `/gdpr_me`

Reply:
Yep, like it